code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/NotificationBuilderTests.kt

```diff
@@ -0,0 +1,300 @@
+        notificationManagerCompat = mockkClass(NotificationManagerCompat::class)
+        every { NotificationManagerCompat.from(any()) } returns notificationManagerCompat
+        every { notificationManagerCompat.cancel(any()) } returns Unit
+        every { alarmManager.canScheduleExactAlarms() } returns true
+        every { context.getSystemService(Context.NOTIFICATION_SERVICE) } returns notificationManager
+        every { context.getSystemService(Context.ALARM_SERVICE) } returns alarmManager
+        every { context.packageName } answers { callOriginal() }
+        every { application.applicationContext } returns context
+        ServiceProvider.getInstance().appContextService.setApplication(application)
+    }
+
+    @Test
+    fun `NotificationBuilder version values matches the expected version`() {
+        val version = NotificationBuilder.version()
+
+        assertEquals("3.0.0", version)
+    }
+
+    @Test
+    fun `constructNotificationBuilder given a map with required data should return a NotificationCompat Builder`() {
+        val mapData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        val notificationBuilder = NotificationBuilder.constructNotificationBuilder(
+            mapData,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertNotNull(notificationBuilder)
+    }
+
+    @Test(expected = NotificationConstructionFailedException::class)
+    fun `constructNotificationBuilder given a map with required data but no context is available should throw an exception`() {
+        setNullContext()
+        val mapData = MockAEPPushTemplateDataProvider.getMockedDataMapWithRequiredData()
+        val notificationBuilder = NotificationBuilder.constructNotificationBuilder(
+            mapData,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertNull(notificationBuilder)
+    }
+
+    @Test(expected = NotificationConstructionFailedException::class)
+    fun `constructNotificationBuilder given a map with empty data should thrown an exception`() {
+        val mapData = emptyMap<String, String>()
+        val notificationBuilder = NotificationBuilder.constructNotificationBuilder(
+            mapData,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertNull(notificationBuilder)
+    }
+
+    @Test
+    fun `constructNotificationBuilder given an intent with required data should return a NotificationCompat Builder`() {
+        val bundle = MockAEPPushTemplateDataProvider.getMockedBundleWithRequiredData()
+        val intent = Intent()
+        intent.action = "mockAction"
+        intent.putExtras(bundle)
+        val notificationBuilder = NotificationBuilder.constructNotificationBuilder(
+            intent,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+
+        assertNotNull(notificationBuilder)
```

Comment:
1. nit: Can we verify the type of `notificationBuilder` rather than just checking for not null
2. Can we validate that the data being sent through intent or mapData is being set properly in the NotificationBuilder

Reply:
i'll get these verifications added in a future pr